packages/wonder-blocks-data/src/hooks/__tests__/use-cached-effect.test.js

```diff
@@ -449,59 +616,158 @@ describe("#useCachedEffect", () => {
             rerender({requestId: "ID2"});
 
             // Assert
-            expect(result.current).toStrictEqual(Status.loading());
+            expect(result.current[0]).toStrictEqual(Status.loading());
         });
 
-        it("should trigger render when request is fulfilled and onResultChanged is undefined", async () => {
-            // Arrange
-            const response = Promise.resolve("DATA");
-            const fakeHandler = jest.fn().mockReturnValue(response);
+        it.each(
+            Array.from(FetchPolicy.members()).filter(
+                (e) => e !== FetchPolicy.CacheOnly,
+            ),
+        )(
+            "should trigger render when request is fulfilled and onResultChanged is undefined for FetchPolicy.%s",
+            async (fetchPolicy) => {
+                // Arrange
+                const response = Promise.resolve("DATA");
+                const fakeHandler = jest.fn().mockReturnValue(response);
+                let renderCount = 0;
+                const Component = React.memo(() => {
+                    useCachedEffect("ID", fakeHandler, {fetchPolicy});
+                    renderCount++;
+                    return <div>Hello :)</div>;
+                });
 
-            // Act
-            const {result} = clientRenderHook(() =>
-                useCachedEffect("ID", fakeHandler),
-            );
-            await act((): Promise<mixed> => response);
+                // Act
+                render(<Component />);
+                await reactAct((): Promise<mixed> => response);
 
-            // Assert
-            expect(result.current).toStrictEqual(Status.success("DATA"));
-        });
+                // Assert
+                expect(renderCount).toBe(2);
+            },
+        );
 
-        it("should not trigger render when request is fulfilled and onResultChanged is defined", async () => {
-            // Arrange
-            const response = Promise.resolve("DATA");
-            const fakeHandler = jest.fn().mockReturnValue(response);
+        it.each(
+            Array.from(FetchPolicy.members()).filter(
+                (e) => e !== FetchPolicy.CacheOnly,
+            ),
+        )(
+            "should trigger render once per inflight request being fulfilled and onResultChanged is undefined for FetchPolicy.%s",
+            async (fetchPolicy) => {
+                // Arrange
+                const response = Promise.resolve("DATA");
+                const fakeHandler = jest.fn().mockReturnValue(response);
+                let renderCount = 0;
+                const Component = React.memo(() => {
+                    const [, refetch] = useCachedEffect("ID", fakeHandler, {
+                        fetchPolicy,
+                    });
+                    React.useEffect(() => {
+                        refetch();
+                        refetch();
+                        refetch();
+                        refetch();
```

Comment:
These get collapsed to a single request right?

Reply:
Exactly. That's what the test is asserting.